pandas/_libs/tslibs/timestamps.pyx

```diff
@@ -342,7 +342,7 @@ cdef class _Timestamp(ABCTimestamp):
 
         elif is_array(other):
             # avoid recursion error GH#15183
-            if other.dtype.kind == "M":
+            if other.dtype.kind == "ME":
```

Comment:
we shouldn't change this one - if it's `dtype.kind`, then `'M'` means `datetime` (I think this comes from numpy:
```python
In [23]: np.datetime64('2020-01-01 00:00').dtype.kind
Out[23]: 'M'
```
)